### PR TITLE
TIP-713: Remove product

### PR DESCRIPTION
**Pull request by anaelChardan**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

This PR aims to correct the remove_product behat of the EE but as it is draft related, this fix allow us to create draft for this stuff.

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: CHANGELOG-1.8.md [hunk 1]
```diff
@@ -132,6 +132,7 @@
 - Remove services `pim_datagrid.datasource.result_record.hydrator.product` and `pim_datagrid.datasource.result_record.hydrator.associated_product`
     and parameters `pim_datagrid.datasource.result_record.hydrator.product.class` and `pim_datagrid.datasource.result_record.hydrator.associated_product.class`
 - Remove all standard denormalizers classes `Pim\Component\Catalog\Denormalizer\Standard\*` and services `pim_catalog.denormalizer.standard.*`
+- Add argument `$product` from `addValue` method of `Pim\Component\Catalog\Validator\UniqueValueSet`
```
Comment (jjanvier) [inline CHANGELOG-1.8.md]: ProductInterface fqcn
instead of $product

**Review by jjanvier — APPROVED** [review]
gtm once fixed

**Comment by jjanvier** [comment]
(and squash your commits when you merge ;) )

**Review by damien-carcel — APPROVED** [review]

File: CHANGELOG-1.8.md [hunk 2]
```diff
@@ -132,7 +132,7 @@
 - Remove services `pim_datagrid.datasource.result_record.hydrator.product` and `pim_datagrid.datasource.result_record.hydrator.associated_product`
     and parameters `pim_datagrid.datasource.result_record.hydrator.product.class` and `pim_datagrid.datasource.result_record.hydrator.associated_product.class`
 - Remove all standard denormalizers classes `Pim\Component\Catalog\Denormalizer\Standard\*` and services `pim_catalog.denormalizer.standard.*`
-- Add argument `$product` from `addValue` method of `Pim\Component\Catalog\Validator\UniqueValueSet`
+- Add argument `Pim\Component\Catalog\Model\ProductInterface` from `addValue` method of `Pim\Component\Catalog\Validator\UniqueValueSet`
```
Comment (damien-carcel) [inline CHANGELOG-1.8.md]: to, not from

Comment (anaelChardan) [reply CHANGELOG-1.8.md]: Ok :)

**Pull request merged**